examples/llm/inference/inference.py

```diff
@@ -109,11 +36,55 @@ def get_mosaicgpt_inference_model(checkpoint_yaml_path: str, tokenizer):
 
     checkpoint = torch.load(ckpt_load_path, map_location='cpu')
 
-    model.load_state_dict(checkpoint['state']['model'], strict=True)
+    if 'state' in checkpoint.keys():
+        # it's a full training checkpoint
+        model.load_state_dict(checkpoint['state']['model'], strict=True)
+    else:
+        # it's a weights-only checkpoint
+        model.load_state_dict(checkpoint, strict=True)
 
-    model.cuda()
-    model.eval()
+    if not model.tokenizer.pad_token_id:
```

Comment:
```suggestion
    if model.tokenizer.pad_token_id is not None:
```
in case the pad token id is 0

Reply:
Done